docs/resources/kubernetes_agent_worker.md

```diff
@@ -0,0 +1,67 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "octopusdeploy_kubernetes_agent_worker Resource - terraform-provider-octopusdeploy"
+subcategory: ""
+description: |-
+  This resource manages Kubernetes agent workers in Octopus Deploy.
+---
+
+# octopusdeploy_kubernetes_agent_worker (Resource)
+
+This resource manages Kubernetes agent workers in Octopus Deploy.
+
+## Example Usage
+
+```terraform
+resource "octopusdeploy_kubernetes_agent_worker" "minimal" {
+  name         = "agent-minimal"
+  worker_pools = ["worker-pools-1"]
+  thumbprint   = "96203ED84246201C26A2F4360D7CBC36AC1D232D"
+  uri          = "poll://kcxzcv2fpsxkn6tk9u6d/"
+}
+
+resource "octopusdeploy_kubernetes_agent_worker" "optionals" {
+  name         = "agent-optionals"
+  worker_pools = ["worker-pools-1", "worker-pools-3"]
+  thumbprint   = "96203ED84246201C26A2F4360D7CBC36AC1D232D"
+  uri          = "poll://kcxzcv2fpsxkn6tk9u6d/"
+  machine_policy_id = "machinepolicies-1"
+  upgrade_locked    = true
+  is_disabled       = true
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) The name of this resource.
+- `thumbprint` (String) The thumbprint of the Kubernetes agent's certificate used by server to verify the identity of the agent. This is the same thumbprint that was used when installing the agent.
```

Comment:
Most other targets have thumbprint as optional. This is probably required for the agent vs other targets but just want to raise it to double check this is correct.

Reply:
It is required here because of the way the Kubernetes agent is installed as a polling Tentacle. Normally the agent sends its own thumbprint to Octopus server during installation where it registers itself as a target/worker. However, in the case of Terraform the resource in Octopus server is being created first so the exact thumbprint must be known ahead of time.